src/metrics/BusFactor.ts

```diff
@@ -8,19 +16,97 @@ export class BusFactor extends Metric {
+     *               number of commits by largest contributor, and total commits. 
+     */
     async calculateScore(): Promise<void> {
 
+        //start timer 
         this.startTimer();
         
-        //checks existence of cloned repository
+        //stats/contributors endpoint 
+        let ep = this.url.getBaseAPI() + "/stats/contributors";
+    
+        
+        //metrics values need (explicitly typed or else will be interpreted as boolean)
+        //hc stands for 'highest contributor', defined as individual with most commits, not necessarily most lines
+        let total_commits: number = 0; 
+        let hc_commits = 0; 
+        let total_lines = 0; 
+        let hc_lines = 0;
+
+        let author:string = " "
+        try{
+            //make api call to ep to return container of objects.
+            const response = await axios.get(ep, {headers: {'Authorization': `token ${process.env.GITHUB_TOKEN}`}}); 
+            
+            if(!response || !response.data || !response.data.length){
+                console.log('data is null');
+                this.score = -1; 
+                return;
+            }
+            
+            // console.log(typeof(response)); 
+            
+            //iterate through each contributor data
+            const data = response.data; 
+            data.forEach((contributor : any) =>{
+                
+                //add to total commits
+                total_commits += contributor.total; 
+
+                //get highest contributor (change if needed)
+                if(contributor.total > hc_commits){
+                    hc_commits = contributor.total; 
+                    author = contributor.author.login; //record this to determine whether to change hc_lines
+                    hc_lines = 0; //reset highest contributor lines to be overwritten below
+                }
 
+                //get total lines
+                let individual_contribution = 0;
+                contributor.weeks.forEach((week : any) => {
+                    individual_contribution += week.a + week.d; 
+                });
 
+                total_lines += individual_contribution;
 
+                
+                if ((author == contributor.author.login) || (author == ' ' && individual_contribution > hc_lines)){
+                    hc_lines = individual_contribution;
+                }
+            }); 
+
+        }catch(Error){ 
+            Logger.logDebug("Bus Factor: Error fetching bus factor data from repository\n" + Error);
+            this.score = -1; 
+            this.endTimer(); 
+            return;
+        }
+
+        // console.log(`\nlargest contributor:${author}\nlines altered: ${hc_lines}/${total_lines}\ncommits:${hc_commits}/${total_commits}`);
+
+        //there are certain github repo settings that omit data
+        if(!total_lines && total_commits){
```

Comment:
Does this check for if there are no lines and no commits

Reply:
checked in line 47 if response or response.data was empty/null